### DISCUSSION TITLE
How to access the data generated by a function call

### DISCUSSION THREAD

**Discussion by saurabhgssingh** [microsoft/autogen#1490] [Q&A]
### Describe the issue

I have a python function that generates a dataframe upon execution. But the dataframe or any outputs generated by a function call doesnt get recorded in the reply function.
Is there a way to record the outputs and use them

function example:
def read_df(path):
      return pd.read_csv(path)

**Comment by ekzhu** [comment]
You can save have the data frame saved in a global variable. Otherwise, a string representing the data frame is in the chat_messages field of the agent
